src/ui/actions/app.ts

```diff
@@ -85,6 +85,9 @@ export type SetHoveredLineNumberLocation = Action<"set_hovered_line_number_locat
   location: Location | null;
 };
 export type SetIsNodePickerActive = Action<"set_is_node_picker_active"> & { active: boolean };
+export type SetIsNodePickerInitializing = Action<"set_is_node_picker_initializing"> & {
```

Comment:
:speech_balloon: As an FYI for future reference: this whole pattern of hand-defining TS types for action objects and unioning them together is actually something that the Redux team specifically advises _against_:

https://redux.js.org/usage/usage-with-typescript#avoid-action-type-unions

As we slowly migrate the Redux logic over to Redux Toolkit, that can go away - RTK's `createSlice` generates correctly-typed action creators for you automatically.

But, it's the pattern that's in this codebase already, so it doesn't _hurt_ to add this.

Reply:
Thanks for the added info.

I'm mostly trying to avoid thinking too hard about our Redux setup and how it feels to work with it until after you've made the changes you're working on.